cloudwatch/cloudwatch.go

```diff
@@ -365,26 +362,33 @@ func (output *OutputPlugin) describeLogStreams(name string, nextToken *string) (
 
 // getStreamName attempts to return the correct stream name for the corresponding tag and record.
 func (output *OutputPlugin) getStreamName(tag string, record map[interface{}]interface{}) (name string) {
+	var err error
+
 	// Create a dynamic log stream name based on the provided log key from the config.
-	if output.logStreamKeyName != "" {
-		for k, v := range record {
-			if recordKey, _ := k.(string); recordKey == output.logStreamKeyName {
-				name, _ = v.(string)
-				break
-			}
-		}
+	if output.logStreamPrefix != "" {
+		return output.logStreamPrefix + tag
 	}
 
-	switch {
-	case name != "" && output.logStreamPrefix != "":
-		return output.logStreamPrefix + tag + "." + name
-	case output.logStreamPrefix != "":
-		return output.logStreamPrefix + tag
-	case name != "":
-		return name
-	default:
-		return output.logStreamName
+	// Add the tag to the record so it can be parsed out in the template.
+	split := strings.SplitN(tag, ".", 2)
+	record["TAG"] = tag
+	record["TAG0"] = split[0]
+	record["TAG1"] = split[1]
+
+	if name, err = digTags(record, output.logStreamName); err != nil {
+		// If a user gets this error, they need to fix their log_stream_name template to make it go away. Simple.
+		logrus.Errorf("[cloudwatch %d] parsing template: '%s': %v", output.PluginInstanceID, output.logStreamName, err)
+	}
+
+	delete(record, "TAG")
+	delete(record, "TAG1")
+	delete(record, "TAG2")
```

Comment:
Am I missing something? In line 374-376, we added record for "TAG", "TAG0", and "TAG1". But here, we are removing "TAG", "TAG1", and "TAG2". What's the matter with "TAG2"?

Reply:
I think you're looking at an old commit.

Reply:
Oh, yes. Sorry, never mind.